stdlib/compat.ncl

```diff
@@ -0,0 +1,12 @@
+{
+compat | doc "Nix compatibility layer. should not be used by Nickel program."
+= {
+  with: Array {_: Dyn} -> Str -> Dyn
+  = fun envs field => (
+    array.fold (fun prev current =>
+      if record.has_field field current
+      then current
+      else prev
```

Comment:
```suggestion
    array.fold (fun current prev =>
      if record.has_field field prev
      then prev
      else current
```

I think you swapped the two. I was confused at first, because given the definition of `fold`, I thought this code would incorrectly select the oldest environment with "field" (the first one in the list), instead of the latest. But actually it is correct.

But then, you are testing n times if the field is inside, even if you already know it is. I would do something like:

```
    array.fold (fun current acc =>
      if !acc.found && record.has_field current field
      then {value = current."%{field"}", set = true}
      else acc
```

Then you have to project `.value` instead of `."%{field}"` and use a starting value `{value = null, found = false}` or just `{found = false}`.

Reply:
Yes, better like this :) but I have to revert the array right?

Reply:
No, because you first look into the accumulator "have the field been already found in the tail" ? If not, set this one as found, otherwise, use the one of the tail. Because this is a right fold, this should be in the right direction.